modules/distributor/distributor.go

```diff
@@ -378,11 +393,20 @@ func (d *Distributor) sendToIngestersViaBytes(ctx context.Context, userID string
 			Traces:     make([]tempopb.PreallocBytes, len(indexes)),
 			Ids:        make([]tempopb.PreallocBytes, len(indexes)),
 			SearchData: nil, // support for flatbuffer/v2 search has been removed. todo: cleanup the proto
+			SpanCounts: make([]tempopb.PreallocBytes, len(indexes)),
 		}
 
 		for i, j := range indexes {
 			req.Traces[i].Slice = marshalledTraces[j][0:]
 			req.Ids[i].Slice = traces[j].id
+
+			// convert span count int to byte : int => string => byte
+			spanCount, err := tempo_util.HexStringToTraceID(strconv.Itoa(traces[j].spanCount))
```

Comment:
definitely shouldn't be going int -> string -> byte. that's a large performance penalty to go from int -> byte